src/editor/UI/Dock.re

```diff
@@ -0,0 +1,31 @@
+open Revery_UI;
+open Revery.UI.Components;
+open Oni_Model;
+
+let component = React.component("Dock");
+
+let dockItemSize = 20;
+
+let icon =
+  Style.[fontFamily("FontAwesome5FreeSolid.otf"), fontSize(dockItemSize)];
+
+let button = Style.[marginVertical(8)];
+
+let createElement = (~children as _, ~state: State.t, ()) =>
+  component(hooks => {
+    let bg = state.theme.colors.editorLineNumberBackground;
+    (
+      hooks,
+      <View
+        style=Style.[
+          flexGrow(1),
+          top(0),
+          bottom(0),
+          backgroundColor(bg),
+          alignItems(`Center),
+        ]>
+        <Clickable style=button> <Text text={||} style=icon /> </Clickable>
```

Comment:
Putting the Unicode character directly code is hard for me to read (it doesn't show up in GH UI, or my default font on Windows).

Would you mind switching to use the `<FontIcon />` component instead? That has an `icon` parameter where you can specify the code directly. An example usage is here: https://github.com/onivim/oni2/blob/c637fc3994c9c4e223f67a51f7d2b9a4df73f469/src/editor/UI/Tab.re#L129

Bonus: You don't need to set a style to pick up the font awesome font, it's already there by default.

Reply:
@bryphe I've converted these, I did see that component but opted against these for a few reasons I think are worth raising.

Currently that component's api requires `icon=int` and the font awesome website gives unicode strings like `f002`, took some digging to find that it needed, `IconTheme.IconDefinition.parseId` to get the `int` value I needed.

You could generalise that but then the api of the component is too permissive either it optionally takes `icon=int` and `iconStr=string` and under the hood does the parsing of the id etc but then it can be passed with a potentially empty value i.e. it no longer demands only one correct thing, since both would have to be options.

Also I guess it does depend on a developer's font but this way means that scanning what icon each of these is rendering is much harder. If we had a group of these and wanted to change one, the only chance you'd have would be to look up the codes? or change stuff till you found the right one :man_shrugging:

Reply:
> Currently that component's api requires icon=int and the font awesome website gives unicode strings like f002, took some digging to find that it needed, IconTheme.IconDefinition.parseId to get the int value I needed.

Ah, you actually shouldn't need anything special - you can just use `0xf002` directly to use it. OCaml supports specifying numbers by their hex notation if you prefix with `0x`: https://caml.inria.fr/pub/docs/manual-ocaml/lex.html

There should be no need for `IconDefinition.parseId` since there's no reason to use a string... you can just use: 
```
<FontIcon icon={0xf002} />
```

> You could generalise that but then the api of the component is too permissive either it optionally takes icon=int and iconStr=string and under the hood does the parsing of the id etc but then it can be passed with a potentially empty value i.e. it no longer demands only one correct thing, since both would have to be options.

I don't see the latest commit, so I may be missing something - but I don't understand why the `<FontIcon />` would need to take either an int or a string.

 One other thing thats related is there is a file that has these values as constants (so you can use `FontAwesome.circle` instead of `0xf111`): https://github.com/onivim/oni2/blob/ad4ca70447c11f1302f2fd1045f0b564f173489d/src/editor/UI/FontAwesome.re#L7

Would be great to add those icons to that list - and then it would be even clearer:
```
<FontIcon icon={FontAwesome.circle} />
```

(as long as the identifier names are in sync with the FontAwesome names)

> Also I guess it does depend on a developer's font but this way means that scanning what icon each of these is rendering is much harder.

Unfortunately I just see this in both GitHub or in the code:

![image](https://user-images.githubusercontent.com/13532591/55921201-a4470400-5bb0-11e9-9338-50f5f3d74c58.png)

![image](https://user-images.githubusercontent.com/13532591/55921416-7f06c580-5bb1-11e9-8ad3-da9e1fd94f0e.png)

...not too useful for me ☹️ 

> If we had a group of these and wanted to change one, the only chance you'd have would be to look up the codes? or change stuff till you found the right one 

Does having them stored as constants help here? I think, at least once we have language support hooked up, it could be helpful because we could do a 'find-all references' for `FontAwesome.circle`, etc.

Reply:
Ah I wasn't aware of the constants file or that we could use a `0x` prefix that solves the issue then :+1:

Reply:
I've included the full list of icons in #242, you can pick up that commit if you wish.

Reply:
@romgrk - that's awesome! Thank you!